tests/openedx_tagging/core/tagging/test_models.py

```diff
@@ -119,16 +138,16 @@ class TestModelTagTaxonomy(TestTagTaxonomyMixin, TestCase):
 
     def test_system_defined(self):
         assert not self.taxonomy.system_defined
-        assert self.system_taxonomy.system_defined
+        assert self.system_taxonomy.cast().system_defined
```

Comment:
Needing to remember this `.cast` is likely to be a problem in the future.

My recommendation (for you to consider in the future!) is that you add a decorator like `@auto_cast` to the base class properties like `.system_defined` that will detect when the developer forgot to `.cast()` and either auto-cast or throw an exception. Probably the latter to be more predictable.